### PR TITLE
Added validation for Input text for dates

### LINKED ISSUE #66: Accepting invalid DATE, YEAR while in text mode and returning last selected date.
https://user-images.githubusercontent.com/43469683/113103777-57391e00-921d-11eb-8462-ac2380a7794d.mov

https://user-images.githubusercontent.com/43469683/113103812-60c28600-921d-11eb-9ea5-62d72bc41690.mov

### PR DESCRIPTION
**Pull request by SantoshVarunk**
closes #66

### CODE REVIEW

**Comment by chakrihacker** [comment]
Any update on this @RichardLindhout ??

File: src/Date/CalendarEdit.tsx [hunk 1]
```diff
@@ -165,11 +176,26 @@ function CalendarInputPure(
...
+      monthLength[1] = 29
+
     if (Number.isNaN(day) || Number.isNaN(year) || Number.isNaN(month)) {
       setError(inputFormat)
       return
     }
 
+    if (month === 0 || month > 12) {
+      setError('MM should be in range of 1-12')
+      onChange(undefined)
```
Comment (RichardLindhout) [inline src/Date/CalendarEdit.tsx:189]: I don't think we should change to undefined only show the error

File: src/Date/CalendarEdit.tsx [hunk 2]
```diff
@@ -165,11 +176,26 @@ function CalendarInputPure(
...
+      return
+    }
+
+    if (day === 0 || day > monthLength[month - 1]) {
+      setError('DD should be in range of 1-' + monthLength[month - 1])
+      onChange(undefined)
```
Comment (RichardLindhout) [inline src/Date/CalendarEdit.tsx:195]: I don't think we should change to undefined only show the error

File: src/Date/CalendarEdit.tsx [hunk 3]
```diff
@@ -165,11 +176,26 @@ function CalendarInputPure(
     const year = Number(date.slice(yearIndex, yearIndex + 4))
     const month = Number(date.slice(monthIndex, monthIndex + 2))
 
+    if (year % 400 === 0 || (year % 100 !== 0 && year % 4 === 0))
+      monthLength[1] = 29
```
Comment (RichardLindhout) [inline src/Date/CalendarEdit.tsx:180]: Why 29?

File: src/Date/DatePickerModalHeader.tsx [hunk 4]
```diff
@@ -32,7 +33,11 @@ export default function DatePickerModalHeader(
               color={color}
             />
             <Appbar.Content title={''} />
-            <Button color={color} onPress={props.onSave}>
+            <Button
+              color={color}
+              onPress={props.onSave}
+              disabled={!props.state.date}
```
Comment (RichardLindhout) [inline src/Date/DatePickerModalHeader.tsx:39]: I don't want to disable buttons, I think the error should be sufficient enough

File: src/Date/CalendarEdit.tsx [hunk 5]
```diff
@@ -154,8 +154,19 @@ function CalendarInputPure(
       .replace('10', 'MM')
       .replace('01', 'DD')
   }, [formatter])
+  const [formattedValue, setFormattedValue] = React.useState(null)
+  const monthLength = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+  React.useEffect(() => {
```
Comment (RichardLindhout) [inline src/Date/CalendarEdit.tsx:160]: Can we move this to the onChange handler instead of useEffect?

**Review by RichardLindhout — CHANGES_REQUESTED** [review]
@ SantoshVarunk Thanks for your PR I still think some changes are required before we can merge this

File: src/Date/CalendarEdit.tsx [hunk 6]
```diff
@@ -165,11 +176,26 @@ function CalendarInputPure(
...
+    if (year % 400 === 0 || (year % 100 !== 0 && year % 4 === 0))
+      monthLength[1] = 29
+
     if (Number.isNaN(day) || Number.isNaN(year) || Number.isNaN(month)) {
       setError(inputFormat)
       return
     }
 
+    if (month === 0 || month > 12) {
+      setError('MM should be in range of 1-12')
```
Comment (RichardLindhout) [inline src/Date/CalendarEdit.tsx:188]: error's should be configurable and be added to examples + readme so users from other countries can configure the right errors

File: src/Date/DatePickerModalHeader.tsx (hunk 4 above)
Comment (RichardLindhout) [reply src/Date/DatePickerModalHeader.tsx:39]: Backup: https://uxdesign.cc/why-you-shouldnt-include-disabled-interaction-elements-in-your-design-system-76a2d4307faf

File: src/Date/CalendarEdit.tsx (hunk 3 above)
Comment (SantoshVarunk) [reply src/Date/CalendarEdit.tsx:180]: Leap year 29 days. index 1 will be Feb month. Above if conditions is verify if it is leap year.

**Comment by iM-GeeKy** [comment]
I'm going to close due to lack of movement. If it is still applicable, please update and we can readdress it.

**Pull request closed without merging**